src/components/structures/RoomView.js

```diff
@@ -748,6 +754,21 @@ export default createReactClass({
             }
         }
     },
+    onEventDecrypted(ev) {
+    if (!SettingsStore.getValue('dontShowChatEffects')) {
+        if (ev.isBeingDecrypted() || ev.isDecryptionFailure() ||
+            this.state.room.getUnreadNotificationCount() === 0) return;
+        this.handleConfetti(ev);
+    }
+    },
+    handleConfetti(ev) {
```

Comment:
this method is only being called in `onEventDecrypted` so it'll only work in e2ee rooms